internals/overlord/checkstate/manager.go

```diff
@@ -387,6 +433,69 @@ type checker interface {
 	check(ctx context.Context) error
 }
 
+func (c *checkData) writeMetrics(writer metrics.Writer) error {
```

Comment:
This should be singular:
```suggestion
func (c *checkData) writeMetric(writer metrics.Writer) error {
```